biogeochem/EDPatchDynamicsMod.F90

```diff
@@ -3217,155 +3217,90 @@ subroutine InsertPatch(currentSite, newPatch)
-          ! patch land use label number, the new patch must be inserted somewhere
-          ! in between oldest and youngest
-          do while(associated(currentPatch) .and. .not. found_landuselabel_match)
-             if (currentPatch%land_use_label .eq. newPatch%land_use_label .or. &
-                 currentPatch%land_use_label .lt. newPatch%land_use_label) then
-                found_landuselabel_match = .true.
-             else
-                currentPatch => currentPatch%older
-             endif
-          end do
+             currentPatch%younger => newPatch
 
-          ! In the case where we've found a landuse label matching the new patch label
-          ! insert the newPatch will as the youngest patch for that land use type.
-          if (associated(currentPatch)) then
+             patch_inserted = .true.
+          endif
+          currentPatch => currentPatch%older
+       end do
+    end if
 
-             newPatch%older    => currentPatch
-             newPatch%younger  => currentPatch%younger
-             currentPatch%younger%older => newPatch
-             currentPatch%younger       => newPatch
+    if ( .not. patch_inserted) then
+       ! something has gone wrong. abort.
+       write(fates_log(),*) 'something has gone wrong in the patch insertion, because no place to put the new patch was found' 
+          call endrun(msg=errMsg(sourcefile, __LINE__))
+    end if
 
-          else
+  end subroutine InsertPatch
 
-             ! In the case were we get to the end, the new patch
-             ! must be numerically the smallest, so put it at the oldest position
-             newPatch%older    => null()
-             newPatch%younger  => currentSite%oldest_patch
-             currentSite%oldest_patch%older   => newPatch
-             currentSite%oldest_patch   => newPatch
+  ! =====================================================================================
 
-          endif
+  function get_pseudo_patch_age(CurrentPatch) result(pseudo_age)
+    
+    ! Purpose: we want to sort the patches in a way that takes into account both their
+    ! continuous and categorical variables. Calculate a pseudo age that does this, by taking
+    ! the integer labels, multiplying these by large numbers, and adding to the continuous age.
 
-       ! Option 4 - always add the new patch as the youngest regardless of land use label
-       case (age_order_only)
-          ! Set the current patch to the youngest patch
-          newPatch%older                     => currentSite%youngest_patch
-          newPatch%younger                   => null()
-          currentSite%youngest_patch%younger => newPatch
-          currentSite%youngest_patch         => newPatch
-       end select
-    end if
+    type (fates_patch_type), intent(in), pointer :: CurrentPatch
+    real(r8)            :: pseudo_age    
+    real(r8), parameter :: max_actual_age = 1.e4  ! hard to imagine a patch older than 10,000 years
```

Comment:
Alternative: we could bump up the max_actual_age even more 1.e5, and then use the max_pft constant along with it:

pseudo_age = max_actual_age * max_pft * land_use_label + pft_label*max_actual_age + age

max_pft will be less than 100

Reply:
@ckoven , what is the range of values on CurrentPatch%nocomp_pft_label? Would it be less than max_pft?  What does it default to when we don't have nocomp turned on?